### ISSUE TITLE
OnPush + ViewContainerRef.createComponent() + ng-container breaks

### ISSUE THREAD

**Issue by zip-fa** [angular/angular#53731]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

Yes

### Description

Hi. This code will not render component:

host-cmp.ts:

```
import {
  ChangeDetectionStrategy,
  ViewContainerRef,
  Component,
  ViewChild,
  AfterViewInit,
} from '@angular/core';
import { TestCmp } from './test-cmp';

@Component({
  selector: 'host-cmp',
  template: '<ng-container #testForVcr />',
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush, // THIS breaks
})
export class HostCmp implements AfterViewInit {
  @ViewChild('testForVcr', { read: ViewContainerRef })
  private readonly vcr!: ViewContainerRef;

  ngAfterViewInit(): void {
    this.vcr.createComponent(TestCmp);
  }
}
```

child-cmp.ts:

```
import {
  ChangeDetectionStrategy,
  Component,
  OnInit,
  signal,
} from '@angular/core';

@Component({
  selector: 'test-cmp',
  template: 'hi from test cmp: {{ testSig() }}',
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TestCmp implements OnInit {
  public readonly testSig = signal<'123' | '456'>('456');

  constructor() {
    console.log('test cmp construct');
  }

  ngOnInit(): void {
    this.testSig.set('123');
    console.log('test cmp init');
  }
}
```

`ngOnInit` doesn't get called also, while constructor() calls.
If i remove OnPush from parent-cmp everything works as expected

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-zvdbww?file=src%2Ftest-cmp.ts

### Please provide the environment you discovered this bug in (run `ng version`)

```true
latest
```

**Comment by JeanMeche** [comment]
Hi, isn't this a dupe of #52928 ?

**Comment by zip-fa** [comment]
Indeed! Sorry for duplicating

**Comment by JeanMeche** [comment]
Yeah, confirmed #53022 fixed it. The "fix" has landed in `17.1.0-next.5` ! 

Since this is a breaking change it will fully land in v18. 

In the mean time, you can call `ɵsetEnsureDirtyViewsAreAlwaysReachable(true);` before bootstraping you app the enable the fix.
